### ISSUE TITLE
bug: Interpolated strings with different literal text content wrongly considered equal

### ISSUE THREAD

**Issue by sproott** [fsprojects/FSharpLint#856]
## Description

The built-in hint `if x then y else y ===> y` triggers on `if/then/else` expressions where both branches are interpolated strings (`$"..."`) that contain the same fill expressions but **different literal text segments**. As the literal text segments are different, the branches should not be considered equivalent.

## Repro

`fsharplint.json` (relevant excerpt — using the default hints shipped with FSharpLint):

```json
{
  "hints": {
    "add": [
      "if x then y else y ===> y"
    ]
  }
}
```

Source:

```fsharp
let label (p: int) (currentPage: int) =
    if p = currentPage then $"Page {p}" else $"Goto page {p}"
```

## Expected

No warning. The two branches have different interpolated sources (`"Page 3"` vs `"Goto page 3"`).

## Actual

FSharpLint reports the `if x then y else y ===> y` hint and suggests replacing the whole expression with one of the branches, which would change behavior.

## Notes

- Making the fill expressions different from each other, e.g. `if p = currentPage then $"Page {p}" else $"Goto page {currentPage}"` suppresses the warning, suggesting only the fill expressions are compared for equality, ignoring the surrounding text
- Replacing the interpolated strings with `sprintf` (`sprintf "Page %d" p` / `sprintf "Goto page %d" p`) suppresses the warning, as a workaround
- Reproduced with FSharpLint version: `0.26.10`
- .NET SDK: `10.0.108`
